analyzer/queries/queries.go

```diff
@@ -1147,4 +1147,107 @@ var (
     WHERE
       (evs.abi_parsed_at IS NULL OR evs.abi_parsed_at < abi_contracts.verification_info_downloaded_at)
     LIMIT $2`
+
+	RuntimeConsensusAccountTransactionStatusUpdate = `
+    -- Transaction should have been in the previous successful round.
+    -- We currently don't have the runtime-block status field in the DB, so find
+    -- the previous successful round by taking the first round which has at least one transaction.
+    WITH tx_round AS (
+      SELECT
+        MAX(rt.round) AS round
+      FROM chain.runtime_transactions AS rt
+      WHERE
+        rt.runtime = $1 AND
+        rt.round < $2
+    ),
+    matched_tx AS (
+      SELECT
+        rt.runtime,
+        rt.round,
+        rt.tx_index
+      FROM chain.runtime_transactions AS rt
+      JOIN chain.runtime_transaction_signers AS rts ON
+        rt.runtime = rts.runtime AND
+        rt.round = rts.round AND
+        rt.tx_index = rts.tx_index
+      WHERE
+        rt.runtime = $1 AND
+        rt.round = (SELECT round FROM tx_round) AND
+        rt.method = $3 AND
+        rts.signer_address = $4 AND
+        rts.signer_index = 0 AND
+        rts.nonce = $5
```

Comment:
> Does the (signer, nonce) here uniquely identify the corresponding transaction if there are two consensus.Deposit txs in the same block?

Yes, since there can be only one transaction per signer in a block.

> the confirmation event is guaranteed to be emitted in block n+1 and not any future block, correct?

Close, but no. The event will be in the first next _successfully processed_ block. In practice this will be the next block in most cases, but we cannot rely on it. We currently don't track the block status (TODO: will add an issue for this), so in `tx_round` subquery we search for the previous block with at least one transaction (that should be the block where the corresponding transaction was sent).